var/spack/repos/builtin/packages/ecp-data-vis-sdk/package.py

```diff
@@ -108,6 +108,9 @@ class EcpDataVisSdk(BundlePackage, CudaPackage, ROCmPackage):
     dav_sdk_depends_on("faodel+shared+mpi network=libfabric", when="+faodel", propagate=["hdf5"])
 
     dav_sdk_depends_on("hdf5@1.12: +shared+mpi", when="+hdf5", propagate=["fortran"])
+    dav_sdk_depends_on("hdf5-vol-async", when="+hdf5")
+    dav_sdk_depends_on("hdf5-vol-cache", when="+hdf5")
```

Comment:
@kwryankrattiger `hdf5-vol-cache` depends on `hdf5-vol-async`, but since the package already depends on this just above it AFAICT we don't need to do anything special.  Correct?

Reply:
correct, no additional actions are required.

Everything listed as a `dav_sdk_depends_on` is only saying "this is a package explicitly provided by the SDK".

If packages have co-dependent requirements with other SDK packages, those are handled at the level of the specific packages, not by the SDK. The SDK should not attempt to duplicate requirements better expressed at the package level.

The only time the SDK interferes with inter-dependencies is to propagate SDK variants or to handle higher order/decoupled dependency resolution. An example of decoupled dependency requirements is HDF5 and Ascent. Ascent requires conduit, but using a Conduit with HDF5 >= v1.8 requires `conduit ~hdf5_compat`. The concretizer struggles to make this happen on it's own, so it is specified at the SDK level.